core/trino-main/src/main/java/io/trino/metadata/DisabledSystemSecurityMetadata.java

```diff
@@ -195,6 +195,10 @@ public Optional<Identity> getFunctionRunAsIdentity(Session session, CatalogSchem
         return Optional.empty();
     }
 
+    @Override
+    public void setFunctionOwner(Session session, CatalogSchemaFunctionName view, TrinoPrincipal principal)
```

Comment:
I think you need to follow the convention here. Instead of saying what you want to do on event, you should simply emit the event and handle what you want in the implementation.

So you should emit events, when function is created, dropped or renamed. Like for any other object.